File: src/main/frontend/src/app/models/dto/response/questionResponse.ts
import {QuestionType} from "../../enums/questionType";
import {AnswerResponse} from "./answerResponse";

export interface QuestionResponse {
  id: number,
  questionText: string,
  questionType: QuestionType,
  order: number,
  inputMinLimit: number | null,
  inputMaxLimit: number | null,
  answers: AnswerResponse[]
}

File: src/main/backend/src/main/java/io/server/backend/dto/response/QuestionResponse.java
package io.server.backend.dto.response;

import io.server.backend.model.enums.QuestionType;

import java.util.List;

public record QuestionResponse(
        Long id,
        String questionText,
        QuestionType questionType,
        Integer order,
        Integer inputMinLimit,
        Integer inputMaxLimit,
        List<AnswerResponse> answers
) { }
